### PR TITLE
DTSCCI-684 Refactor RespondToClaimCallbackHandler

### PR DESCRIPTION
**Pull request by Jefex7**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DTSCCI-684

### Change description ###

Reduce Cyclomatic Complexity of Handler RespondToClaimCallbackHandler

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/PopulateRespondentCopyObjects.java [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+    public CallbackResponse execute(CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+        LocalDateTime dateTime = LocalDateTime.now();
+
+        CallbackResponse errorResponse = checkSubmitError(callbackParams, caseData, dateTime);
+        if (errorResponse != null) {
+            return errorResponse;
+        }
+
+        var isRespondent1 = findOutIsRespondent1(callbackParams);
```
Comment (gmikerlemos) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/PopulateRespondentCopyObjects.java:79]: I would try not using var, I know the latest java interactions allow this but I believe the strong typification in Java is on of its best features so I'd stick to strongly typing variables.

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/PopulateRespondentCopyObjects.java [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+        // Show error message if defendant tries to submit a response after deadline has passed
+        if (isResponseSubmissionPassedDeadline(callbackParams, caseData, dateTime)) {
+            return createErrorResponse("You cannot submit a response now as you have passed your deadline");
+        }
+
+        return null;
+    }
+
+    private YesOrNo findOutIsRespondent1(CallbackParams callbackParams) {
```
Comment (gmikerlemos) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/PopulateRespondentCopyObjects.java:108]: Allow my pettiness, but findOut is not a good name to check something. I'd set it either as a isRespondent1Present, checkRespondent1YesOrNo or simply checkRespondent1.

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/PopulateRespondentCopyObjects.java [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+                                                CallbackParams callbackParams,
+                                                LocalDateTime dateTime,
+                                                LocalDateTime respondent2ResponseDeadline) {
+        return solicitorRepresentsOnlyOneOfRespondents(callbackParams, RESPONDENTSOLICITORTWO)
+            && caseData.getRespondent2ResponseDate() == null
+            && respondent2ResponseDeadline != null
+            && dateTime.toLocalDate().isAfter(respondent2ResponseDeadline.toLocalDate());
+    }
+
+    private boolean solicitorRepresentsOnlyOneOrBothRespondents(CallbackParams callbackParams, CaseRole caseRole) {
```
Comment (gmikerlemos) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/PopulateRespondentCopyObjects.java:212]: please use standard method naming when returning boolean. isSomething().

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/PopulateRespondentCopyObjects.java [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+            userInfo.getUid(),
+            caseRole
+        );
+    }
+
+    private boolean solicitorRepresentsOnlyOneOfRespondents(CallbackParams callbackParams, CaseRole caseRole) {
+        return solicitorRepresentsOnlyOneOrBothRespondents(callbackParams, caseRole);
+    }
+
+    private List<LocationRefData> fetchLocationData(CallbackParams callbackParams) {
```
Comment (gmikerlemos) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/PopulateRespondentCopyObjects.java:227]: again, fetch can be used but the standard is getSomething, setSomething, isSomething

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/PopulateRespondentCopyObjects.java [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+        UserInfo userInfo = userService.getUserInfo(callbackParams.getParams().get(BEARER_TOKEN).toString());
+        return stateFlowEngine.evaluate(caseData).isFlagSet(TWO_RESPONDENT_REPRESENTATIVES)
+            && coreCaseUserService.userHasCaseRole(
+            caseData.getCcdCaseReference().toString(),
+            userInfo.getUid(),
+            caseRole
+        );
+    }
+
+    private boolean solicitorRepresentsOnlyOneOfRespondents(CallbackParams callbackParams, CaseRole caseRole) {
```
Comment (gmikerlemos) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/PopulateRespondentCopyObjects.java:223]: I didn't get the reason for this method, calling another method without anything else. Why not calling the method directly?

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/PopulateRespondentCopyObjects.java [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+            .flatMap(applicantCourt -> locations.stream()
+                .filter(locationRefData -> applicantCourt.equals(locationRefData.getCourtLocationCode()))
+                .findFirst())
+            .ifPresent(locationRefData -> requestedCourt1
+                .otherPartyPreferredSite(locationRefData.getCourtLocationCode()
+                                             + " " + locationRefData.getSiteName()));
+        return requestedCourt1.build();
+    }
+
+    private boolean hasDefendantAlreadySubmitted(CallbackParams callbackParams, CaseData caseData) {
```
Comment (gmikerlemos) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/PopulateRespondentCopyObjects.java:166]: boolean isSomething()

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/PopulateRespondentCopyObjects.java [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+                .respondent2Copy(caseData.getRespondent2())
+                .respondent2DetailsForClaimDetailsTab(updatedCaseData.build().getRespondent2()
+                                                          .toBuilder().flags(null).build());
+        }
+        return updatedCaseData;
+    }
+
+    private RequestedCourt createRequestedCourt(List<LocationRefData> locations, CaseData caseData) {
+        DynamicList courtLocationList = courtLocationUtils.getLocationsFromList(locations);
+        RequestedCourt.RequestedCourtBuilder requestedCourt1 = RequestedCourt.builder();
```
Comment (gmikerlemos) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/PopulateRespondentCopyObjects.java:152]: is there any reason for calling it requestedCourt1 if apparently there isn't a second one (requestedCourt2)?

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/PopulateRespondentCopyObjects.java [hunk 8]
```diff
@@ -0,0 +1,231 @@
...
+        RequestedCourt requestedCourt1 = createRequestedCourt(locations, caseData);
+
+        CaseData.CaseDataBuilder updatedCaseData = updateCaseData(caseData, isRespondent1, requestedCourt1);
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(updatedCaseData.build().toMap(objectMapper))
+            .build();
+    }
+
+    private CallbackResponse checkSubmitError(CallbackParams callbackParams,
```
Comment (gmikerlemos) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/PopulateRespondentCopyObjects.java:92]: it should be called, getErrorResponse() or something like it, getSomething

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/PopulateRespondentCopyObjects.java [hunk 9]
```diff
@@ -0,0 +1,231 @@
...
+    }
+
+    private CallbackResponse createErrorResponse(String errorMessage) {
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .errors(List.of(errorMessage))
+            .build();
+    }
+
+    private CaseData.CaseDataBuilder updateCaseData(CaseData caseData, YesOrNo isRespondent1, RequestedCourt requestedCourt1) {
+        var updatedCaseData = caseData.toBuilder()
```
Comment (gmikerlemos) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/PopulateRespondentCopyObjects.java:124]: same here with var, and you are creating a new caseData object, you will override the data saved when I believe, what you are trying to do is just update some values, so you shouldn't create a new instance with the toBuilder()

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/ValidateRespondentExperts.java [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+
+        }
+        if (hasRespondent2DifferentResponseAndExperts(caseData)) {
+            return validateExperts(caseData.getRespondent2DQ());
+        }
+
+        return validateExperts(caseData.getRespondent1DQ());
+    }
+
+    private boolean hasRespondent2DifferentResponseAndExperts(CaseData caseData) {
```
Comment (gmikerlemos) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/ValidateRespondentExperts.java:60]: isSomething method name patter here please

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/ValidateRespondentExperts.java [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+        return (respondent2HasSameLegalRep(caseData)
+            && (caseData.getRespondentResponseIsSame() != null && caseData.getRespondentResponseIsSame() == NO)
+            && (caseData.getRespondent2DQ() != null && caseData.getRespondent2DQ().getRespondent2DQExperts() != null));
+    }
+
+    private boolean solicitorRepresentsOnlyOneOfRespondents(CallbackParams callbackParams, CaseRole caseRole) {
```
Comment (gmikerlemos) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/ValidateRespondentExperts.java:66]: same with the method that doesn't do anything

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/ValidateRespondentExperts.java [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+        return solicitorRepresentsOnlyOneOrBothRespondents(callbackParams, caseRole);
+    }
+
+    private boolean solicitorRepresentsOnlyOneOrBothRespondents(CallbackParams callbackParams, CaseRole caseRole) {
```
Comment (gmikerlemos) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/ValidateRespondentExperts.java:70]: isSomething

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/ValidateRespondentExperts.java [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+        UserInfo userInfo = userService.getUserInfo(callbackParams.getParams().get(BEARER_TOKEN).toString());
+        return stateFlowEngine.evaluate(caseData).isFlagSet(TWO_RESPONDENT_REPRESENTATIVES)
+            && coreCaseUserService.userHasCaseRole(
+            caseData.getCcdCaseReference().toString(),
+            userInfo.getUid(),
+            caseRole
+        );
+    }
+
+    private boolean respondent2HasSameLegalRep(CaseData caseData) {
```
Comment (gmikerlemos) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/ValidateRespondentExperts.java:81]: same, isSomething

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/ValidateRespondentWitnesses.java [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    private boolean hasRespondent2DifferentResponseAndWitnesses(CaseData caseData) {
+        return respondent2HasSameLegalRep(caseData)
+            && caseData.getRespondentResponseIsSame() == NO
+            && caseData.getRespondent2DQ() != null
+            && caseData.getRespondent2DQ().getRespondent2DQWitnesses() != null;
+    }
+
+    private boolean solicitorRepresentsOnlyOneOfRespondents(CallbackParams callbackParams, CaseRole caseRole) {
```
Comment (gmikerlemos) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/ValidateRespondentWitnesses.java:67]: isSomething

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/ValidateRespondentWitnesses.java [hunk 15]
```diff
@@ -0,0 +1,88 @@
...
+        return solicitorRepresentsOnlyOneOrBothRespondents(callbackParams, caseRole);
+    }
+
+    private boolean respondent2HasSameLegalRep(CaseData caseData) {
+        return caseData.getRespondent2SameLegalRepresentative() != null
+            && caseData.getRespondent2SameLegalRepresentative() == YES;
+    }
+
+    private boolean solicitorRepresentsOnlyOneOrBothRespondents(CallbackParams callbackParams, CaseRole caseRole) {
```
Comment (gmikerlemos) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/ValidateRespondentWitnesses.java:76]: first, isSomething
second, the method name could be better, the flag is giving the clue. why not calling it isTwoRespondentRepresentative?

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/ValidateUnavailableDates.java [hunk 16]
```diff
@@ -0,0 +1,92 @@
...
+        this.userService = userService;
+    }
+
+    @Override
+    public CallbackResponse execute(CallbackParams callbackParams) {
+
+        CaseData caseData = callbackParams.getCaseData();
+        Hearing hearing = caseData.getRespondent1DQ().getHearing();
+
+        if (!ONE_V_ONE.equals(MultiPartyScenario.getMultiPartyScenario(caseData))
```
Comment (gmikerlemos) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/ValidateUnavailableDates.java:51]: could be extracted to an isSomething method

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/ValidateUnavailableDates.java [hunk 17]
```diff
@@ -0,0 +1,92 @@
...
+        }
+
+        List<String> errors = unavailableDateValidator.validate(hearing);
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .errors(errors)
+            .build();
+    }
+
+    private boolean hasRespondent2DifferentResponseAndHearing(CaseData caseData) {
```
Comment (gmikerlemos) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/ValidateUnavailableDates.java:64]: please change to IsSomething pattern

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimCallbackHandlerTest.java [hunk 18]
```diff
@@ -102,6 +107,11 @@
 @ExtendWith(SpringExtension.class)
 @SpringBootTest(classes = {
```
Comment (gmikerlemos) [inline src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimCallbackHandlerTest.java:108]: Have we not agreed to remove spring context tests from unit tests?

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertaskstest/PopulateRespondentCopyObjectsTest.java [hunk 19]
```diff
@@ -0,0 +1,186 @@
...
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.civil.callback.CallbackParams.Params.BEARER_TOKEN;
+import static uk.gov.hmcts.reform.civil.enums.CaseRole.RESPONDENTSOLICITORONE;
+import static uk.gov.hmcts.reform.civil.enums.CaseRole.RESPONDENTSOLICITORTWO;
+
+@ExtendWith(SpringExtension.class)
```
Comment (gmikerlemos) [inline src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertaskstest/PopulateRespondentCopyObjectsTest.java:36]: I think you don't need the Spring runner here, if you are not using the spring test facilities. Which would most likely involve crating a spring context then just use MockitoExtension.

**Review by gmikerlemos — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/SetApplicantResponseDeadline.java [hunk 20]
```diff
@@ -0,0 +1,665 @@
...
+    public CallbackResponse execute(CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+
+        CaseData.CaseDataBuilder<?, ?> updatedData = updateRespondentAddresses(caseData);
+
+        LocalDateTime responseDate = time.now();
+        AllocatedTrack allocatedTrack = caseData.getAllocatedTrack();
+        LocalDateTime applicant1Deadline = getApplicant1ResponseDeadline(responseDate, allocatedTrack);
+
+        // 1v2 same legal rep - will respond for both and set applicant 1 response deadline
```
Comment (gmikerlemos) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/SetApplicantResponseDeadline.java:118]: Avoid adding comments to code. Your code should be clean enough and descriptive enough to not make comments necessary

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/SetApplicantResponseDeadline.java [hunk 21]
```diff
@@ -0,0 +1,665 @@
...
+        updateDQPartyIdsIfHmcEnabled(updatedData);
+
+        caseFlagsInitialiser.initialiseCaseFlags(DEFENDANT_RESPONSE, updatedData);
+
+        // casefileview changes need to assign documents into specific folders, this is help determine
+        // which user is "creating" the document and therefore which folder to move the documents
+        // into, when directions order is generated in GenerateDirectionsQuestionnaireCallbackHandler
+        updateDocumentGenerationRespondent2(callbackParams, updatedData, caseData);
+
+        if (getMultiPartyScenario(caseData) == ONE_V_TWO_TWO_LEGAL_REP
```
Comment (gmikerlemos) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/SetApplicantResponseDeadline.java:162]: should be extracted to an isSomething method

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/SetApplicantResponseDeadline.java [hunk 22]
```diff
@@ -0,0 +1,665 @@
...
+        // these documents are added to defendantUploads, if we do not remove/null the original,
+        // case file view will show duplicate documents
+        nullPlaceHolderDocuments(updatedData, caseData);
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(updatedData.build().toMap(objectMapper))
+            .state("AWAITING_APPLICANT_INTENTION")
+            .build();
+    }
+
+    private void handleDefaultResponse(CallbackParams callbackParams,
```
Comment (gmikerlemos) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/SetApplicantResponseDeadline.java:179]: handle, process, etc is a bit generic for a private  method that is doing something very specific (it should be), be more descriptive with method names and you won't need comments.

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/SetApplicantResponseDeadline.java [hunk 23]
```diff
@@ -0,0 +1,665 @@
...
+            }
+        }
+    }
+
+    private boolean applicant2Present(CaseData caseData) {
+        return caseData.getAddApplicant2() != null && caseData.getAddApplicant2() == YES;
+    }
+
+    private void setApplicantDeadlineIfRequired(CaseData caseData, CaseData.CaseDataBuilder<?, ?> updatedData, LocalDateTime applicant1Deadline) {
+        if (respondent2NotPresent(caseData)
```
Comment (gmikerlemos) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/SetApplicantResponseDeadline.java:306]: isSomething method extraction here

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/SetApplicantResponseDeadline.java [hunk 24]
```diff
@@ -0,0 +1,665 @@
...
+
+        // resetting statement of truth to make sure it's empty the next time it appears in the UI.
+        updatedData.uiStatementOfTruth(StatementOfTruth.builder().build());
+        // 1v2 same Solicitor responding to respondents individually
+    }
+
+    private  CaseData.CaseDataBuilder<?, ?> updateRespondentAddresses(CaseData caseData) {
+        if (ofNullable(caseData.getRespondent1Copy()).isPresent()
+            && (caseData.getRespondent1Copy().getPrimaryAddress() == null)) {
+            throw new IllegalArgumentException("Primary Address cannot be empty");
```
Comment (gmikerlemos) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/SetApplicantResponseDeadline.java:408]: This should be checked through CCD definitions if this field is mandatory

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/SetApplicantResponseDeadline.java [hunk 25]
```diff
@@ -0,0 +1,665 @@
+package uk.gov.hmcts.reform.civil.handler.callback.user.task.respondtoclaimcallbackhandlertasks;
```
Comment (gmikerlemos) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/SetApplicantResponseDeadline.java:1]: This class is too big is and is kind of defeating the purpose of the refactor. see if you can split this into more specialised classes. Also please remove all comments.

**Review by gmikerlemos — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/ValidateRespondentExperts.java (hunk 11 above)
Comment (Jefex7) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/ValidateRespondentExperts.java:66]: deleted

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimCallbackHandlerTest.java (hunk 18 above)
Comment (Jefex7) [reply src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimCallbackHandlerTest.java:108]: I believe are Integration tests

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/ValidateRespondentWitnesses.java (hunk 15 above)
Comment (Jefex7) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/ValidateRespondentWitnesses.java]: in the other comments you just mentionned that you wanted the is pattern isSolicitorRepresentingOneOrBothRespondents descriptive enough ?

**Review by nigeldunne — APPROVED** [review]

**Review by ruban72 — DISMISSED** [review]

**Review by gmikerlemos — DISMISSED** [review]

**Pull request closed without merging**